### ISSUE TITLE
[PM-646] improve the expunge operation in monitord in replay mode

### ISSUE THREAD

**Issue by mayani** [pegasus-isi/pegasus#764]
In case of large SCEC MySQL databases, the expunge process ( deleting already populated data in the mysql db for a workflow ) takes a long time, resulting in a spike in the memory usage.

This is because of the way expunge has been implemented in the netlogger SQL Alchemy code.
It triggers individual select statements , the results of which are then invidually deleted.

The Expunge code already does things specific to a database backend
1) for sqlite it just overwrites

Hence, for mysql it is possible to rely on database triggers in the schema definition to do cascaded deletes

**Reporter:** @vahi
**Resolution:** Fixed
**Watchers:**
@scallag
@vahi

**Comment by mayani** [comment]
**Author:** @vahi

When monitord is run in a replay mode, the database is first expunged of all the information related to that workflow. In case, of mysql backend where the same database maybe used to track multiple hierarchal workflows, the expunge operation has to be careful to delete only the relevant entries for the various tables.

In earlier versions, this expunge operation was implemented at OR level in SQLAlchemy that led to lots of select and delete statements to be executed ( one per entry ). This blew up the memory footprint for monitord and prevented the workflow population in case of large databases. For 4.1, we changed the schema to add cascaded delete clauses, and set the passive delete option to true in SQLAlchemy.
